src/EventStore.Core/Telemetry/Instant.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace EventStore.Core.Telemetry;
+
+// this provides stronger typing than just passing a long representing the number of ticks
+// and provides us a place to change the resolution and size if long ticks is overkill.
+public struct Instant {
+	private static readonly double _secondsPerTick = 1 / (double)Stopwatch.Frequency;
+
+	private static readonly Stopwatch _stopwatch = Stopwatch.StartNew();
```

Comment:
Instead of the `static readonly`-field you could use `Stopwatch.GetTimestamp()` (the static method on Stopwatch).
* avoids the fields access
* the other methods on SW also call `GetTimestamp` internally -- so should be minimally faster

Reply:
oh yeah, thanks!